src/EFCore.MySql/Storage/Internal/MySqlTypeMappingSource.cs

```diff
@@ -298,6 +324,16 @@ private RelationalTypeMapping FindRawMapping(RelationalTypeMappingInfo mappingIn
                             return _timeStampOffset6;
                         }
                     }
+                    else if (storeTypeName.Equals("tinyint(1)", StringComparison.OrdinalIgnoreCase)
+                             && _options.ConnectionSettings.TreatTinyAsBoolean)
+                    {
+                        return _tinyint1;
+                    }
+                    else if (storeTypeName.Equals("bit(1)", StringComparison.OrdinalIgnoreCase)
+                             && !_options.ConnectionSettings.TreatTinyAsBoolean)
+                    {
+                        return _bit1;
+                    }
```

Comment:
Why not use the StoreType of the _tinyint1 and _bit1 field?

Reply:
That's a valid point. I replaced all of them with field references now.

Reply:
Thinking about this a bit more, it should be possible to move most of the dynamic type checking out of `FindRawMapping()` and into `Initialize()`.